Modules/Core/Common/test/itkImageRegionRangeGTest.cxx

```diff
@@ -0,0 +1,598 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+ // First include the header file to be tested:
+#include "itkImageRegionRange.h"
+
+#include "itkImage.h"
+#include "itkImageBufferRange.h"
+#include "itkImageRegionIterator.h"
+#include "itkMacro.h" // For itkNotUsed.
+#include "itkRangeGTestUtilities.h"
+#include "itkVectorImage.h"
+
+#include <gtest/gtest.h>
+#include <algorithm>  // For std::reverse_copy, std::equal, etc.
+#include <numeric>  // For std::inner_product
+#include <type_traits>  // For std::is_reference.
+
+// Test template instantiations for various ImageDimension values, and const Image:
+template class itk::Experimental::ImageRegionRange<itk::Image<short, 1>>;
+template class itk::Experimental::ImageRegionRange<itk::Image<short, 2>>;
+template class itk::Experimental::ImageRegionRange<itk::Image<short, 3>>;
+template class itk::Experimental::ImageRegionRange<itk::Image<short, 4>>;
+template class itk::Experimental::ImageRegionRange<const itk::Image<short>>;
```

Comment:
Since you are being this comprehensive, maybe you should also add `RGBPixel<uint8_t>` and `Vector<long, 11>` as pixel types.

Reply:
Thanks for the suggestion @dzenanz  I've put it onto my TODO list  `:-)`

Reply:
@dzenanz 
> maybe you should also add `RGBPixel<uint8_t>` and `Vector<long, 11>` as pixel types.

I just did, please check the latest force-push!